torchdata/datapipes/iter/util/rar_archive_reader.py

```diff
@@ -0,0 +1,45 @@
+import io
+import os.path
+from typing import Tuple, Iterator
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils.common import validate_pathname_binary_tuple
+
+
+@functional_datapipe("read_from_rar")
+class RarArchiveReaderIterDataPipe(IterDataPipe[Tuple[str, io.BufferedIOBase]]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, io.BufferedIOBase]]):
+        self._rarfile = self._verify_dependencies()
```

Comment:
Although it is a static method, I don't see any benefit in calling it as such here. What is your reasoning?

Reply:
`_verify_dependencies` does not have `self` in its argument. And `self` makes the method looking like bounded to instance rather than class. A little benefit on convention?